### PR TITLE
issue: 3373882 Fix compilation issue for gcc 13.0.1

### PR DESCRIPTION
**Pull request by igor-ivanov**
## Description
Fix compilation issue for gcc 13.0.1

##### What
Fix compilation

##### Why ?
RM:3373882 

##### How ?
_It is optional but for complex PRs please provide information about the design,
architecture, approach, etc._

## Change type
What kind of change does this PR introduce?
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Tests
- [ ] Other

## Check list
- [ ] Code follows the style de facto guidelines of this project
- [ ] Comments have been inserted in hard to understand places
- [ ] Documentation has been updated (if necessary)
- [ ] Test has been added (if possible)

### CODE REVIEW

**Review by AlexanderGrissik — APPROVED** [review]

**Comment by igor-ivanov** [comment]
bot:retest

File: config/m4/compiler.m4 [hunk 1]
```diff
@@ -143,7 +143,7 @@ case $CC in
                 -Wno-free-nonheap-object "
         CXXFLAGS="$CXXFLAGS -Wshadow -Wall -Wextra -Werror -Wundef \
                 -ffunction-sections -fdata-sections -Wsequence-point -pipe -Winit-self -Wmissing-include-dirs \
-                -Wno-free-nonheap-object "
+                -Wno-free-nonheap-object -Wno-overloaded-virtual "
```
Comment (AlexanderGrissik) [inline config/m4/compiler.m4:146]: Is it really OK to avoid this warning? Wouldn't it be better to fix it?

Comment (igor-ivanov) [reply config/m4/compiler.m4:146]: I did not find a way to fix them in simple manner.
Task for fixing this and others option can be planned.

**Comment by igor-ivanov** [comment]
bot:retest

**Comment by igor-ivanov** [comment]
bot:retest

**Comment by igor-ivanov** [comment]
bot:retest

**Comment by igor-ivanov** [comment]
bot:retest

**Pull request merged**